### ISSUE TITLE
CWE ID 377

### ISSUE THREAD

**Issue by ptahchiev** [raphw/byte-buddy#784]
Hello we are using VeraCode to find potential security problems in our code and the tool claims it has found a very serious security problem with byte-buddy:
```
byte-buddy-agent-1.9.16.jar net/.../agent/ByteBuddyAgent.java 636
byte-buddy-agent-1.9.16.jar net/.../agent/ByteBuddyAgent.java 1456 (edited) 
```
with the following description:
```
Description
Creating and using insecure temporary files can leave application and system data vulnerable to attack. In particular,
file names created by the tmpnam family of functions can be easily guessed by an attacker. If an attacker can predict
the filename and create a malicious collision, he may be able to manipulate the behavior of the application.
Effort to Fix: 2 - Implementation error. Fix is approx. 6-50 lines of code. 1 day to fix.
Recommendations
Ensure that unpredictable names are used for temporary files and that files are created in a secure directory with
appropriate permissions. Using mkstemp() is a reasonably safe way to create temporary files. It will attempt to create
and open a unique file based on a filename template provided by the user, combined with a series of randomly
generated characters. Note that mkstemp() is safe if only the descriptor is used and the returned filename is not used
in a subsequent function call with extra privileges. Using mkstemp() does not completely eliminate race conditions but
does provide better protection than other methods.
```
It seems like the problem is here:
https://github.com/raphw/byte-buddy/blob/master/byte-buddy-agent/src/main/java/net/bytebuddy/agent/ByteBuddyAgent.java#L636

**Comment by akkamburov** [comment]
Also reported with Very High Severity is vulnerability "CWE ID 78" at ByteBuddyAgent.java line 659
...
if (new ProcessBuilder(System.getProperty(JAVA_HOME)
...
reported in byte-buddy-agent-1.9.16, but apparently available in the latest versions too.

**Comment by raphw** [comment]
I think this has been an issue in old versions of Java 6 and on Java 5 where the temporary file was not created randomly if the file already existed. In modern Java versions, this should no longer be an issue.

**Comment by akkamburov** [comment]
we're using jdk versions 8 and 11

**Comment by raphw** [comment]
Then this really should not affect you.

**Comment by akkamburov** [comment]
Issue solved!
Turns out Veracode tool runs its tests against the java version that each given library is compiled against. And since the official releases of ByteBuddy are compiled with java 5 it tests it with 5.
I pulled the latest version byte-buddy-1.10.4, built it (w/o any changes of course) with my jdk11 
(though internally it seems to use 7:
"ByteBuddy.class: compiled Java class data, version 51.0 (Java 1.7)")
so Veracode tests it against Java7 and it passes alright.
